src/HDF5.jl

```diff
@@ -589,22 +590,27 @@ See also
 """
 function create_dataset(
     parent::Union{File,Group},
-    path::AbstractString,
+    path::Union{AbstractString, Nothing},
     dtype::Datatype,
     dspace::Dataspace;
     dcpl::DatasetCreateProperties = DatasetCreateProperties(),
     dxpl::DatasetTransferProperties = DatasetTransferProperties(),
     dapl::DatasetAccessProperties = DatasetAccessProperties(),
     pv...
 )
-    haskey(parent, path) && error("cannot create dataset: object \"", path, "\" already exists at ", name(parent))
+    path !== nothing && haskey(parent, path) && error("cannot create dataset: object \"", path, "\" already exists at ", name(parent))
     pv = setproperties!(dcpl,dxpl,dapl; pv...)
     isempty(pv) || error("invalid keyword options")
-    Dataset(API.h5d_create(parent, path, dtype, dspace, _link_properties(path), dcpl, dapl), file(parent), dxpl)
+    if path isa Nothing
```

Comment:
```suggestion
    if isnothing(path)
```
would be more canonical

Reply:
https://docs.julialang.org/en/v1/base/base/#Base.isnothing is only available with Julia 1.1. Have we broken compat with Julia 1.0?

Reply:
https://github.com/JuliaIO/HDF5.jl/blob/master/Project.toml#L17

Reply:
Yes